[Security Solution][Endpoint] Fix Respond console input parser so that bare command arguments are given a blank string

paul-tavares opened the pull request:
## Summary

- Fix console command input parser so that if an argument is used with with no value at all, it is given by default of empty string. Example: `foo --one` would have the `one` parsed value be set to `one: [ true ]`

Issue found in https://github.com/elastic/kibana/issues/137570#issuecomment-1212767744

<img width="935" alt="image" src="https://user-images.githubusercontent.com/56442535/186762475-51d43640-09b5-4f0a-a4a6-9ae533da6e45.png">

elasticmachine commented:
Pinging @elastic/security-onboarding-and-lifecycle-mgt (Team:Onboarding and Lifecycle Mgt)

pzl reviewed (approved):
the end result is fine. good catch by qas on that bug

i always shake my head at the various ways to use falsy values in JS and assign them meanings. Here, semantically, `cmd --foo ""` would be treated the same as `cmd --foo` where that's not _necessarily_ true for real flag parsers. But I doubt we will ever get _there_

paul-tavares commented:
@joeypoon , @pzl - I fixed up the types and adjusted the arguments parsing code so that if an argument is provided with no value, then a boolean `true` will be assign to it as the value instead of empty string. When you get a chance, can you give this another review 🙏

joeypoon commented on x-pack/plugins/security_solution/public/management/components/endpoint_responder/endpoint_response_actions_console_commands.ts:
```diff
@@ -23,7 +23,7 @@ import type {
 
 const emptyArgumentValidator = (argData: ParsedArgData): true | string => {
-  if (argData?.length > 0 && argData[0]?.trim().length > 0) {
+  if (argData?.length > 0 && 'string' === typeof argData[0] && argData[0]?.trim().length > 0) {
```
nit: I feel like it's more standard to reverse this `typeof argData[0] === 'string'`.

joeypoon reviewed (approved):
non-blocking nit, otherwise 👌.
